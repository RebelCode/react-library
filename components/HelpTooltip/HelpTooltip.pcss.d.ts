export const root: string
export const tooltip: string
export const tooltipContainer: string
export const tooltipContent: string
export const icon: string
export const iconVisible: string
export const iconNotvisible: string
interface Namespace {
	"root": string,
	"tooltip": string,
	"tooltipContainer": string,
	"tooltip-container": string,
	"tooltipContent": string,
	"tooltip-content": string,
	"icon": string,
	"iconVisible": string,
	"icon-visible": string,
	"iconNotvisible": string,
	"icon-notvisible": string,
}
declare const helpTooltipPcss: Namespace
export default helpTooltipPcss
