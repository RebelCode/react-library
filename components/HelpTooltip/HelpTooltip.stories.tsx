import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import HelpTooltip from "./index";

export default {
  title: "Components/HelpTooltip",
  component: HelpTooltip,
  parameters: {
    // More on Story layout: https://storybook.js.org/docs/react/configure/story-layout
    layout: "fullscreen",
  },
} as ComponentMeta<typeof HelpTooltip>;

const Template: ComponentStory<typeof HelpTooltip> = (args) => (
  <HelpTooltip {...args} />
);

export const Help = Template.bind({});
Help.args = {
  maxWidth: 200,
  children: <>One, Two. Three</>
};

