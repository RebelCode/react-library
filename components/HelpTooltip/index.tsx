import React, {ReactNode} from "react";
import css from "./HelpTooltip.pcss";
import {Dashicon} from "../Dashicon";
import Tooltip from "../Tooltip";

type Props = {
    maxWidth?: number;
    children: ReactNode;
}

export default function HelpTooltip({maxWidth, children}: Props) {
    maxWidth = maxWidth ?? 300;

    const [isTooltipVisible, setIsTooltipVisible] = React.useState(false);

    const handleMouseOver = () => setIsTooltipVisible(true);
    const handleMouseOut = () => setIsTooltipVisible(false);

    const tooltipTheme = {
        content: css.tooltipContent,
        container: css.tooltipContainer,
    };

    return (
        <div className={css.root}>
            <Tooltip visible={isTooltipVisible} theme={tooltipTheme}>
                {
                    ({ref}) => (
                        <span ref={ref}
                              className={css.icon `${isTooltipVisible ? css.iconVisible : css.iconNotvisible}`}
                              onMouseEnter={handleMouseOver}
                              onMouseLeave={handleMouseOut}>
                            <Dashicon icon="info" />
                        </span>
                    )
                }

                <div style={{maxWidth: maxWidth + "px"}}>{children}</div>
            </Tooltip>
        </div>
    );
};
