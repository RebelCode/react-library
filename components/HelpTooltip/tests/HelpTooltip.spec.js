import React from "react";
import { render, cleanup } from "@testing-library/react";
import HelpTooltip from "../index";

afterEach(cleanup);

describe("HelpTooltip", () => {
  const Comp = (
    <HelpTooltip maxWidth={200}>
      One, Two. Three
    </HelpTooltip>
  );

  it("should take a snapshot", () => {
    const { asFragment } = render(Comp);

    expect(asFragment(Comp)).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { queryByLabelText } = render(Comp);

    expect(queryByLabelText("HelpTooltip")).toBeTruthy();
  });
});
