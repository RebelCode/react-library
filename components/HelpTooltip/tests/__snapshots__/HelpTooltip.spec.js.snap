// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpTooltip should take a snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="HelpTooltip"
    class="root"
  >
    <span
      class="icon iconNotvisible"
    >
      <span
        aria-label="Dashicon"
        class="dashicons dashicons-info"
      />
    </span>
  </div>
</DocumentFragment>
`;
